libraries/ts/src/margin/pool/pool.ts

```diff
@@ -1313,7 +1318,7 @@ export class Pool {
       ),
       // adapterMetadata: new PublicKey("DUheebnZrHMGzEMbs9FpPFTkbmVdZnyW92CVwrYd3aGa"),
       adapterInstruction: await this.programs.marginSwap.methods
-        .marginSwap(swapAmount.lamports, minAmountOut.lamports)
+        .marginSwap(changeKind.changeKind, changeKind.value, minAmountOut.lamports)
```

Comment:
This may allow anchor to correctly serialize it if you were having problems
```suggestion
        .marginSwap(changeKind.changeKind.asParam(), changeKind.value, minAmountOut.lamports)
```